### PR TITLE
Remove params to maintain backward-compatibility

### LINKED ISSUE #35: Remove "service_refresh_file_list" parameter and "remove_file_list" from init constructor
"service_refresh_file_list" parameter and "remove_file_list" parameter is not required to be passed to the init's constructor. Furthermore, passing these parameters will break backward compatibility.

### PR DESCRIPTION
**Pull request by vidurananayakkara**
Resolves #35

### CODE REVIEW

File: manifests/params.pp [hunk 1]
```diff
@@ -31,6 +31,8 @@
     $java_source_file         = hiera('wso2::java::source_file')
     $java_user                = hiera('wso2::java::user')
     $java_group               = hiera('wso2::java::group')
+    $service_refresh_file_list = hiera('wso2::service_refresh_file_list', undef)
+    $remove_file_list         = hiera('wso2::remove_file_list', undef)
```
Comment (imesh) [inline manifests/params.pp]: It might be better to align the assignment operator to the same vertical line to be aligned with the previous standard.

**Review by imesh — CHANGES_REQUESTED** [review]

**Review by chamilad — CHANGES_REQUESTED** [review]
@vidurananayakkara Wouldn't modules like APIM which are already released with these parameters in the base module call [1] break after this change? IMO if the intention is to bring all the modules to a single `wso2base` contract, then it might be better to keep these parameters and do releases for the product modules.

[1] - https://github.com/wso2/puppet-apim/blob/v2.1.0/wso2am_runtime/manifests/init.pp#L133

**Comment by vidurananayakkara** [comment]
@chamilad Yes, existing modules (like APIM) which are using [1] will break. However, there are other modules (like ESB) which haven't incorporated [1] and will break. We either have to,

1. Add [1] to those modules which don't have [1] 
2. Remove [1] from modules which have [1] following a common pattern.

Since the approach in this PR encourages a common pattern to introduce a new parameter, I would suggest the approach in this PR.

WDYT?

[1] https://github.com/wso2/puppet-apim/blob/v2.1.0/wso2am_runtime/manifests/init.pp#L133

**Comment by imesh** [comment]
I think Chamila has a point! It would be better not to remove remove_file_list parameter as it has been used in released product puppet modules.

**Comment by vidurananayakkara** [comment]
Reverted changes for [1]

[1] https://github.com/wso2/puppet-apim/blob/v2.1.0/wso2am_runtime/manifests/init.pp#L133

**Review by imesh — APPROVED** [review]

**Comment by chamilad** [comment]
So if we are pushing for product module release approach, IMO we might have to keep this addition and a few others that were added in the past and then removed, and do a major release for the `wso2base` module. Then for the product modules that were released so far, we should do a round of testing and do releases with possibly a version level lower than patch number.

AFAIK the modules so far released are,
1. wso2/puppet-am
2. wso2/puppet-is (to be released)
3. wso2/puppet-ei
4. wso2/puppet-das

I'm not sure of the effort this would take though since we need to verify these modules in a Master/Agent setup. However, I feel this is the best approach we have for now to bring all the modules to one `wso2base` level. 

WDYT?

**Comment by chamilad** [comment]
I had an offline discussion with @vidurananayakkara during which the above points were discussed in detail. The final conclusion was to do a minor/patch release for `wso2base` for the moment with the changes in this PR and then follow up with a `puppet-is` release that coincides with the `wso2base` release. This is to immediately wrap up the changes done recently regarding Secure Vault implementation support in Puppet.

After this is done, the effort to bring all the product Puppet modules to a single base should be prioritized. This is especially of importance since, after the Secure Vault implementation support changes, none of the released product Puppet modules would work correctly.

**Review by chamilad — APPROVED** [review]

**Pull request merged**